controller/environment.go

```diff
@@ -75,8 +68,9 @@ func (c *EnvironmentController) Create(ctx *app.CreateEnvironmentContext) error
 	if reqEnv == nil {
 		return app.JSONErrorResponse(ctx, errors.NewBadParameterError("data", nil).Expected("not nil"))
 	}
+
 	spaceID := ctx.SpaceID
-	err = c.checkSpaceExist(ctx, spaceID.String())
+	err = c.authClient.CheckSpaceScope(ctx, spaceID.String(), "manage")
```

Comment:
I would add "manage", "contribute", and "view" as constants to common `auth` package.

Reply:
string_literal has its own benefit like code is more readable and no need to open constant definition.  many places we set http_headers like `Authorization`, `If-Modified-Since`, `Content-Type` with string_literal rather constant.